### PR TITLE
feat(datasource/kubernetes-api): add kubernetes-api datasource

### PR DESCRIPTION
**Pull request by JamieMagee**
## Changes

Adds the Kubernetes API datasource

TODO:
- [x] Test on a real repository
	- https://github.com/JamieMagee/renovate-kubernetes-api/pull/3
	- https://github.com/JamieMagee/renovate-kubernetes-api/pull/4
- [x] Documentation for the datasource

## Context

#5972 

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/datasource/kubernetes-api/mapping.ts [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  // https://kubernetes.io/docs/reference/using-api/deprecation-guide/#daemonset-v116
+  DaemonSet: ['extensions/v1beta1', 'apps/v1beta2', 'apps/v1'],
+  // https://kubernetes.io/docs/reference/using-api/deprecation-guide/#deployment-v116
+  Deployment: ['extensions/v1beta1', 'apps/v1beta1', 'apps/v1beta2', 'apps/v1'],
+  // https://kubernetes.io/docs/reference/using-api/deprecation-guide/#statefulset-v116
+  StatefulSet: ['apps/v1beta1', 'apps/v1beta2', 'apps/v1'],
+  // https://kubernetes.io/docs/reference/using-api/deprecation-guide/#replicaset-v116
+  ReplicaSet: ['extensions/v1beta1', 'apps/v1beta1', 'apps/v1beta2', 'apps/v1'],
+  // https://kubernetes.io/docs/reference/using-api/deprecation-guide/#psp-v116
+  PodSecurityPolicy: ['extensions/v1beta1', 'policy/v1beta1'],
```
Comment (viceice) [inline lib/modules/datasource/kubernetes-api/mapping.ts:14]: those will be removed in V1.25

File: lib/modules/datasource/kubernetes-api/mapping.ts [hunk 2]
```diff
@@ -0,0 +1,118 @@
+export const KubernetesApiVersions: Record<string, string[]> = {
```
Comment (viceice) [inline lib/modules/datasource/kubernetes-api/mapping.ts:1]: should we move this to a json file, so we can some tea in future update this automatically?

**Review by viceice — COMMENTED** [review]
needs test / coverage fixes

File: lib/modules/datasource/kubernetes-api/mapping.ts (hunk 1 above)
Comment (JamieMagee) [reply lib/modules/datasource/kubernetes-api/mapping.ts:14]: True, but upgrading to `policy/v1beta1` is as far as Renovate can go. The migration guide[^1] requires manual intervention.

[^1]: https://kubernetes.io/docs/tasks/configure-pod-container/migrate-from-psp/

File: lib/modules/datasource/kubernetes-api/mapping.ts (hunk 2 above)
Comment (JamieMagee) [reply lib/modules/datasource/kubernetes-api/mapping.ts:1]: I can do, but I've not been able to identify a machine readable source for this information. The deprecation guide is to free form to write a parser for.

Comment (JamieMagee) [reply lib/modules/datasource/kubernetes-api/mapping.ts]: Moved this to `data/kubernetes-api.json5`, but still unable to identify a machine readable source.

**Review by viceice — APPROVED** [review]

**Pull request merged**